test/Microsoft.DotNet.PackageInstall.Tests/EndToEndToolTests.cs

```diff
@@ -500,6 +500,141 @@ static XElement GetToolSettingsFile(string packagePath)
 
         }
 
+        [Fact]
+        public void InstallToolWithHigherFrameworkAsGlobalToolShowsAppropriateError()
+        {
+            var toolPackagesPath = CreateNet99ToolPackage();
+            var testDirectory = _testAssetsManager.CreateTestDirectory();
+            var homeFolder = Path.Combine(testDirectory.Path, "home");
+
+            var result = new DotnetToolCommand(Log, "install", "-g", "Net99Tool", "--add-source", toolPackagesPath)
+                .WithEnvironmentVariables(homeFolder)
```

Comment:
Can you update the WithEnvironmentVariable method so that it also sets NUGET_PACKAGES to a `.nuget\packages` folder under the home folder?  This will make it so that the test package doesn't leak out into a user-wide or repo packages folder (if that were to happen then the tests could end up consuming older versions of the mock package since the version doesn't change).